### PR TITLE
improve code quality

### PR DESCRIPTION
**Pull request by withshubh**
#### Description
This PR fixes a few issues that were affecting the code quality.

#### Summary of changes

- Combine multiple `append`s into a single call
- Fix Yoda conditions
- Fix check for empty string

### CODE REVIEW

File: length_test.go [hunk 1]
```diff
@@ -503,8 +493,7 @@ func TestMsgCompressLengthEscaped(t *testing.T) {
 	msg := new(Msg)
 	msg.Compress = true
 	msg.SetQuestion("www.example.org.", TypeA)
-	msg.Answer = append(msg.Answer, &NS{Hdr: RR_Header{Name: `\000\001\002.example.org.`, Rrtype: TypeNS, Class: ClassINET}, Ns: `ns.\e\x\a\m\p\l\e.org.`})
-	msg.Answer = append(msg.Answer, &NS{Hdr: RR_Header{Name: `www.\e\x\a\m\p\l\e.org.`, Rrtype: TypeNS, Class: ClassINET}, Ns: "ns.example.org."})
+	msg.Answer = append(msg.Answer, &NS{Hdr: RR_Header{Name: `\000\001\002.example.org.`, Rrtype: TypeNS, Class: ClassINET}, Ns: `ns.\e\x\a\m\p\l\e.org.`}, &NS{Hdr: RR_Header{Name: `www.\e\x\a\m\p\l\e.org.`, Rrtype: TypeNS, Class: ClassINET}, Ns: "ns.example.org."})
```
Comment (tmthrgd) [inline length_test.go:496]: The changes in this file make it much harder to read. It’s also only a test so I think revert these.

**Review by tmthrgd — CHANGES_REQUESTED** [review]
Some changes, but otherwise this is good. Thanks.

File: length_test.go (hunk 1 above)
Comment (withshubh) [reply length_test.go]: reverted! :sparkles:

**Comment by withshubh** [comment]
@tmthrgd Done! :sparkling_heart:

**Review by tmthrgd — APPROVED** [review]
LGTM.

**Pull request merged**